internal/server/server.go

```diff
@@ -48,10 +57,54 @@ type StorerGetter interface {
 	Getter
 }
 
-func New(metadataStore Storer, objectStore StorerGetter) *Server {
-	return &Server{metadataStore, objectStore}
+type StoreResponse struct {
+	Gitoid string `json:"gitoid"`
 }
 
+func New(metadataStore Storer, objectStore StorerGetter, cfg *config.Config, sqlClient *ent.Client) Server {
+	r := mux.NewRouter()
+	s := &Server{metadataStore, objectStore, nil}
+
+	// TODO: remove from future version (v0.5.0) endpoint with version
+	r.HandleFunc("/download/{gitoid}", s.GetHandler)
+	r.HandleFunc("/upload", s.StoreHandler)
+	if cfg.EnableGraphql {
+		r.Handle("/query", s.Query(sqlClient))
+		r.Handle("/v1/query", s.Query(sqlClient))
+	}
+
+	r.HandleFunc("/v1/download/{gitoid}", s.GetHandler)
+	r.HandleFunc("/v1/upload", s.StoreHandler)
+	if cfg.GraphqlWebClientEnable {
+		r.Handle("/",
+			playground.Handler("Archivista", "/v1/query"),
+		)
+	}
+	r.PathPrefix("/swagger/").Handler(httpSwagger.WrapHandler)
+	s.router = r
+
+	return *s
+
+}
+
+// @title Archivista API
+// @description Archivista API
+// @version v1
+// @contact.name Archivista Contributors
+// @contact.url https://github.com/in-toto/archivista/issues/new
+// @license Apache 2
+// @license.url https://opensource.org/licenses/Apache-2
+// InitRoutes initializes the HTTP API routes for the server
+func (s *Server) Router() *mux.Router {
+	return s.router
+}
+
+// @Summary Store
+// @Description stores an attestation
+// @Produce  json
+// @Success 200 {object} api.StoreResponse
+// @Tags attestation
+// @Router /v1/upload [post]
 func (s *Server) Store(ctx context.Context, r io.Reader) (api.StoreResponse, error) {
```

Comment:
I think in the `server`, we could rename this function to (`AttestationUpload`), it will be more aligned with the API naming for contributors.
I would suggest also including for future deprecation (`Store`) on the `pkg/API/store.`

I know it looks like a tasty decision, but aligning the endpoint's name with function handlers can be very helpful.

If this idea looks ok, we could also do it for the Get -> Download.

Reply:
I think this is a good idea. It seems like it will be more descriptive and easier to understand.

Reply:
I will add it in a specific commit on this PR.

Reply:
I agree with these changes. Never a bad thing to be more descriptive and explicit.